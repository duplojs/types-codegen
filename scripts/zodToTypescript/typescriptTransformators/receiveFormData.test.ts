import { ZodToTypescript } from "@duplojs/zod-to-typescript";
import { recieveFiles, zod } from "@duplojs/core";
import { receiveFormDataTransformator } from "./receiveFormData";
import { instanceofTransformator } from "./instanceof";

it("receiveFormDataTransformator", () => {
	const zodToTypescript = new ZodToTypescript([
		receiveFormDataTransformator,
		instanceofTransformator,
	]);

	zodToTypescript.append(
		zod.receiveFormData({
			picture: recieveFiles({
				maxSize: 500,
				quantity: 5,
				mimeType: /test/,
			}),
			userId: zod.string(),
			age: zod.number(),
		}),
		"Test",
	);

	expect(
		zodToTypescript.toString(true),
	).toMatchSnapshot();
});
