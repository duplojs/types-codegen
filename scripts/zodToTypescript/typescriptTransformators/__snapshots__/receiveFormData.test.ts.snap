// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`receiveFormDataTransformator 1`] = `
"interface CodegenReceiveFormData<GenericValue extends Record<string, string | string[] | number | number[] | Date | Date[] | File[]>> {
    extractor: (...args: any[]) => Promise<GenericValue>;
}

export { CodegenReceiveFormData };

type Test = CodegenReceiveFormData<{
    picture: File[];
    userId: string;
    age: number;
}>;

export { Test };"
`;
